activerecord/lib/active_record/connection_handling.rb

```diff
@@ -45,16 +45,20 @@ module ConnectionHandling
     # The exceptions AdapterNotSpecified, AdapterNotFound and +ArgumentError+
     # may be returned on an error.
     def establish_connection(spec = nil)
+      raise RuntimeError, "Anonymous class is not allowed." unless name
+
       spec     ||= DEFAULT_ENV.call.to_sym
```

Comment:
The problem is that `spec` can be a hash, which is the actual config. So `spec_name` would not work all the time.